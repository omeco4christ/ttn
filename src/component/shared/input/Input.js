import React from 'react';
import {View} from 'react-native';
import styled from '@emotion/native';
import {
  COLOR_INPUT,
  COLOR_SECONDARY,
  COLOR_WHITE,
} from '../../../constants/styles';
import {TextXs} from '../text/Text';
import {CSS_INPUT} from '../../../styles/styles';

const Input = props => {
  const {error, ...others} = props;
  return (
    <View>
      <InputWrapper
        placeholderTextColor={COLOR_WHITE}
        error={error}
        {...others}
      />
      <TextXs color={COLOR_SECONDARY}>{error}</TextXs>
    </View>
  );
};

const InputWrapper = styled.TextInput`
  height: ${props => (props.multiline ? 'auto' : '44px')};
  border-radius: 22px;
  border-color: ${props => (props.error ? COLOR_SECONDARY : COLOR_INPUT)};
  text-align-vertical: top;
  ${CSS_INPUT};
`;
export default Input;
