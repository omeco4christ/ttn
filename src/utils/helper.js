import dayjs from 'dayjs';
import relativeTime from 'dayjs/plugin/relativeTime';

dayjs.extend(relativeTime);

export const logger = console.log;

export const htmlToText = html => html.replace(/<(.*?)>/g, '');

export const timeFromNow = date => dayjs(date).fromNow();

export const displayDate = date => dayjs(date).format('MMM YYYY');
