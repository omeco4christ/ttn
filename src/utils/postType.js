class PostType {
  constructor() {
    this.types = {
      question: 'Ask Question',
      share_link: 'Share a link',
      forum_post: 'Post to Forum',
    };
  }

  getType(type) {
    return this.types[type];
  }

  keysArray() {
    return Object.keys(this.types);
  }

  questionType() {
    return {key: this.keysArray()[0], value: this.types[this.keysArray()[0]]};
  }

  shareLinkType() {
    return {key: this.keysArray()[1], value: this.types[this.keysArray()[1]]};
  }

  forumType() {
    return {key: this.keysArray()[2], value: this.types[this.keysArray()[2]]};
  }
}

const postType = new PostType();

export default postType;
