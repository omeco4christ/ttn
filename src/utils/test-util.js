import React from 'react';
import renderer from 'react-test-renderer';

const AllTheProviders = ({children}) => {
  return <React.Fragment>{children}</React.Fragment>;
};

const customRender = (ui, options) =>
  renderer(ui, {wrapper: AllTheProviders, ...options});

export {customRender as renderer};
