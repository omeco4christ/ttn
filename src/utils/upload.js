import RNFetchBlob from 'rn-fetch-blob';
import {launchImageLibrary} from 'react-native-image-picker';

export const uploadImageFromStorage = () => {
  return new Promise((resolve, reject) => {
    launchImageLibrary(
      {includeBase64: true, maxHeight: 850, maxWidth: 720, mediaType: 'photo'},
      async response => {
        const image = response.assets[0];
        resolve(image);
      },
    );
  });
};

export const uploadImageToServer = image => {
  return new Promise((resolve, reject) => {
    RNFetchBlob.fetch(
      'POST',
      'https://dev-api.tellit.io/api/v1/upload',
      {
        'Content-Type': 'multipart/form-data',
      },
      [
        {
          name: 'file',
          filename: image.fileName,
          type: image.type,
          data: RNFetchBlob.wrap(image.uri),
        },
      ],
    )
      .then(res => {
        let parsedData = JSON.parse(res.data);
        console.log(parsedData.data);
        if (parsedData) {
          resolve(parsedData.data.url);
        }
      })
      .catch(err => reject(err));
  });
};

export const uploadImageFromStorageToServer = async () => {
  const imageFromStorage = await uploadImageFromStorage();
  return await uploadImageToServer(imageFromStorage);
};
