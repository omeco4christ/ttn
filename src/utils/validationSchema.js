import * as Yup from 'yup';

export const loginSchema = Yup.object({
  username: Yup.string().required('Username or email is required'),
  password: Yup.string().required('Password is required'),
});

export const signupSchema = Yup.object({
  first_name: Yup.string().required('First name is required'),
  last_name: Yup.string().required('Last name is required'),
  username: Yup.string().required('Username is required'),
  phone_number: Yup.number().required('Phone number is required'),
  email: Yup.string().email('Invalid email').required('Email is required'),
  password: Yup.string()
    .min(6, 'Password must be at least 6 character')
    .required('Password is required.'),
  confirm_password: Yup.string().oneOf(
    [Yup.ref('password'), null],
    'Passwords not match',
  ),
});

export const forgetPasswordSchema = Yup.object({
  email: Yup.string().email('Invalid email').required('Email is required'),
});

export const resetPasswordSchema = Yup.object({
  password: Yup.string()
    .min(6, 'Password must be at least 6 character')
    .required('Password is required.'),
  confirm_password: Yup.string().oneOf(
    [Yup.ref('password'), null],
    'Passwords not match',
  ),
});

export const resetPasswordTokenSchema = Yup.object({
  token: Yup.number()
    .required('Token is required')
    .max(9999, 'Token must be 4 length'),
});

export const changePasswordSchema = Yup.object({
  old_password: Yup.string().required('Old password is required.'),
  new_password: Yup.string()
    .min(6, 'Password must be at least 6 character')
    .required('New password is required.'),
  confirm_password: Yup.string().oneOf(
    [Yup.ref('new_password'), null],
    'Passwords not match',
  ),
});

export const forumTopicSchema = Yup.object({
  title: Yup.string().required('Title is required'),
  description: Yup.string().required('Description is required'),
});

export const forumTopicCommentSchema = Yup.object({
  comment: Yup.string().required('Comment is required'),
});
