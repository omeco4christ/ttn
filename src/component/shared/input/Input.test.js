import React from 'react';
import {render} from '../../../utils/test-util';
import Input from './Input';
import {expect} from '@jest/globals';
import renderer from 'react-test-renderer';

describe('Input', () => {
  it('', () => {
    const wrapper = renderer.create(<Input />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
