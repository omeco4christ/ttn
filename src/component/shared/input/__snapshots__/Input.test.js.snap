// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input  1`] = `
<TextInput
  allowFontScaling={true}
  rejectResponderTermination={true}
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "borderBottomLeftRadius": 22,
        "borderBottomRightRadius": 22,
        "borderTopLeftRadius": 22,
        "borderTopRightRadius": 22,
        "height": 44,
        "opacity": 0.2,
      },
      undefined,
    ]
  }
  underlineColorAndroid="transparent"
/>
`;
